src/basic/locale-util.h

```diff
@@ -33,7 +33,7 @@ int get_locales(char ***l);
 bool locale_is_valid(const char *name);
 int locale_is_installed(const char *name);
 
-#define _(String) gettext(String)
+#define _(String) dgettext(GETTEXT_PACKAGE, String)
 #define N_(String) String
 void init_gettext(void);
```

Comment:
Hmm, so I think we should not do localization from library functions. In other words, I don't think the gettext-releated stuff should be in `basic`. Instead, let's move it to `shared`. And then the old definition would be correct.

Reply:
While pam_systemd_home.so is not a library, there is no practical difference: it's a plugin loaded by libpam into address space of libpam clients.  That is, pam_systemd_home.so cannot rely on `textdomain()` and therefore has to use `dgettext()` instead of `gettext()`.

Also, pam_systemd_home.so is not linked with libsystemd libraries, so it doesn't look like it would be quite correct to move `_()` to shared, even though it's just a macro.

Reply:
Indeed.